### PR TITLE
Mmdetection task interface

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
run tests

File: ote/ote/tasks/mmdetection/parameters.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from ote.interfaces.parameters import BaseTaskParameters
+
+
+def is_field(name, t):
+    if hasattr(t, name):
+        return True
+    return False
+
+
+def build_train_parameters(train_kwargs_dict, configs_root):
```
Comment (sovrasov) [inline ote/ote/tasks/mmdetection/parameters.py:12]: Please use `ote.modules.arg_converters.convert_args_to_parameters` here in the next PR

File: ote/ote/tasks/mmdetection/dataset.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import numpy as np
+import mmcv
+from mmdet.datasets import DATASETS, CocoDataset
+from mmdet.datasets.pipelines import Compose
+from pycocotools.coco import COCO
+
+from ote.interfaces.dataset import IDataset
+
+
+class CocoDataSource(IDataset):
```
Comment (sovrasov) [inline ote/ote/tasks/mmdetection/dataset.py:13]: In the future please create `IDetectionDataset` and describe output of getitem and get_annoation to allows 3rd party users look at specs and create a dataset that could be used with the detection task.

**Pull request merged**